src/main/java/org/testng/TestNG.java

```diff
@@ -1394,7 +1394,9 @@ private SuiteRunner createSuiteRunner(XmlSuite xmlSuite) {
     }
 
     for (IReporter r : result.getReporters()) {
-      addListener(r);
+      if (!m_reporters.containsKey(r.getClass())) {
```

Comment:
this is what's i'm trying to avoid, since calling `maybeAddListener` will pring the warning message.